### PR TITLE
Expose SendGridClient internals to test project

### PR DESCRIPTION
**Pull request by aevitas**
# Companion PR for #924  # 

This is a PR that acts as a companion to #924, by allowing tests to be written for `internal` members of the client. Without it, we can either not write unit tests for internal code at all, and forego regression tests, or expose the APIs as public. Neither of these options seems very attractive, hence this PR introduces a third option.

In order for `InternalsVisibleTo` to work, any assembly it exposes its internals too must be strong signed, hence the tests project is now strongly signed with the same key as the SendGridClient project.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the development branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Sign SendGrid.Tests
- Expose SendGridClient internals to SendGrid.Tests

### CODE REVIEW

**Comment by aevitas** [comment]
FWIW, #924 can't be fully merged with proper tests until this PR is merged.

**Comment by thinkingserious** [comment]
Manually rebuild this one and it passed: https://travis-ci.com/github/sendgrid/sendgrid-csharp/builds/217686129

File: src/SendGrid/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+using System.Runtime.InteropServices;
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("377c20e4-2297-488f-933b-fb635c56d8fc")]
+[assembly: InternalsVisibleTo("SendGrid.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100812ec26a66c8e08c790704ac4b46bcc9da9f4bca4da0ec7c06ce6dcd73baeb2c5525f36a237b253e80e16febb4c052f50734d5e1cf3bf478d9c88f0f69df53b47306419182983bc35c33c3bafb5e90b9bd7aa7b9a9da09abe3667d50db891012e077e4b9aefe9799a58222fa67127c230219755d7670073c7463d90cf9e79dba")]
```
Comment (thinkingserious) [inline src/SendGrid/Properties/AssemblyInfo.cs:16]: How was this public key generated?

File: tests/SendGrid.Tests/SendGrid.Tests.csproj [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <AssemblyName>SendGrid.Tests</AssemblyName>
+    <PackageId>SendGrid.Tests</PackageId>
+    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
+    <RuntimeFrameworkVersion>1.0.4</RuntimeFrameworkVersion>
+    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
+    <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
+    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
+    <SignAssembly>true</SignAssembly>
```
Comment (thinkingserious) [inline tests/SendGrid.Tests/SendGrid.Tests.csproj:12]: Does the test file need to be signed? I'm not opposed to it, just want to understand the reasoning here.

**Review by thinkingserious — COMMENTED** [review]
Thank you for the PR @aevitas! Just a few questions.

File: src/SendGrid/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (aevitas) [reply src/SendGrid/Properties/AssemblyInfo.cs:16]: The SendGrid assembly is signed using a strongly named key, it's the `tools/sendgrid-csharp.snk` key already in the repository. I just grabbed the public key token for the SNK.

File: tests/SendGrid.Tests/SendGrid.Tests.csproj (hunk 2 above)
Comment (aevitas) [reply tests/SendGrid.Tests/SendGrid.Tests.csproj:12]: I think if one of the assemblies is signed, this is required. But I'd have to go back and check if it's actually strictly needed -- it's been a while! :)

**Comment by thinkingserious** [comment]
Hello @aevitas,

Could you please address the [test error](https://travis-ci.com/github/sendgrid/sendgrid-csharp/builds/223115609)? Thank you!

Reference

```
/root/.nuget/packages/microsoft.net.test.sdk/16.6.1/build/netcoreapp2.1/Microsoft.NET.Test.Sdk.Program.cs(3,38): error CS0012: The type 'Attribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [/app/tests/SendGrid.Tests/SendGrid.Tests.csproj]
```

**Comment by thinkingserious** [comment]
Thank you @aevitas!!

**Pull request closed without merging**